packages/eas-cli/src/graphql/queries/BuildQuery.ts

```diff
@@ -0,0 +1,75 @@
+import gql from 'graphql-tag';
+
+import { graphqlClient, withErrorHandlingAsync } from '../client';
+import { Build } from '../types/Build';
+
+type Filters = Partial<Pick<Build, 'platform' | 'status'>> & {
+  order?: number;
+  limit?: number;
+};
+
+type ArtifactFragmentType = Pick<Build, 'artifacts'>;
+type PlatformAndArtifactFragmentType = Pick<Build, 'platform' | 'artifacts'>;
+
+export class BuildQuery {
+  static async forArtifactByIdAsync(buildId: string): Promise<PlatformAndArtifactFragmentType> {
```

Comment:
```suggestion
  static async byIdAsync(buildId: string): Promise<PlatformAndArtifactFragmentType> {
```

Reply:
- The function name should match the query name.
- If we're going to call this query in different contexts then I'd prefer one of the following approaches:
   - pass a fragment to the function
   - pass an array of fields/edges to fetch
   - always return the same fragment (this is probably the best option as it simplifies the types)

Reply:
I just went with the 3rd approach, the two former needs some kind of query building and... never mind.